etc/init.d/zfs-functions.in

```diff
@@ -132,33 +173,38 @@ zfs_action()
 #
 zfs_daemon_start()
 {
-	local PIDFILE="$1";	shift
-	local DAEMON_BIN="$1";	shift
-	local DAEMON_ARGS="$*"
+	local pidfile="${1}";		shift
+	local daemon_bin="${1}";	shift
+	local daemon_args="$*"
 
-	if type start-stop-daemon > /dev/null 2>&1 ; then
+	if do_noout type start-stop-daemon
+	then
 		# LSB functions
-		start-stop-daemon --start --quiet --pidfile "$PIDFILE" \
-		    --exec "$DAEMON_BIN" --test > /dev/null || return 1
+		start-stop-daemon --quiet --start --test \
+		    --pidfile "${pidfile}" --exec "${daemon_bin}" || \
+		    return 1
 
-	        start-stop-daemon --start --quiet --exec "$DAEMON_BIN" -- \
-		    $DAEMON_ARGS || return 2
+		start-stop-daemon --quiet --start \
+		    --exec "${daemon_bin}" -- ${daemon_args} || \
+		    return 2
 
 		# On Debian GNU/Linux, there's a 'sendsigs' script that will
 		# kill basically everything quite early and zed is stopped
 		# much later than that. We don't want zed to be among them,
 		# so add the zed pid to list of pids to ignore.
-		if [ -f "$PIDFILE" -a -d /run/sendsigs.omit.d ]
-		then
-			ln -sf "$PIDFILE" /run/sendsigs.omit.d/zed
-		fi
-	elif type daemon > /dev/null 2>&1 ; then
+		[ -f "${pidfile}" -a -d /run/sendsigs.omit.d ] && \
+		    ln -sf "${pidfile}" /run/sendsigs.omit.d/zed
+
+	elif do_noout type daemon
+	then
 	        # Fedora/RedHat functions
-		daemon --pidfile "$PIDFILE" "$DAEMON_BIN" $DAEMON_ARGS
-		return $?
+		daemon --pidfile "${pidfile}" "${daemon_bin}" \
+		    ${daemon_args} || \
+		    return 3
+
 	else
 		# Unsupported
-		return 3
+		return -1
```

Comment:
context ;) I needed 3 for the return four lines above. Increasing to 4 didn't seem right as it is not related to 1-3 but a completely different error. Since -1 isn't posix this returns 255 now.

Reply:
I'm wondering if the return codes have meaning within the init system they're called from? E.g. "3" might *mean* "unsupported" and changing the return code might change the behavior of the init system.